packages/documentation/src/components/Header.jsx

```diff
@@ -1,7 +1,140 @@
+/* eslint-disable react/jsx-key */
 import React from 'react';
-import { Link } from 'gatsby';
+import { graphql, Link, StaticQuery } from 'gatsby';
 import sidebarData from '../sidebar';
 import MobileNav from './MobileNav';
+import { Index } from 'elasticlunr';
+
+const maxDisplayedSearchResults = 5;
+
+function StringWithBoldQuery(props) {
+  const { string, queries } = props;
+  const lowerCasedQueries = queries.map(query => query.toLowerCase());
+  const lowerCasedString = string.toLowerCase();
+
+  const matchedQuery = lowerCasedQueries.find(query =>
+    lowerCasedString.includes(query),
+  );
+  if (!matchedQuery) {
+    return <>{string}</>;
+  }
+
+  const queryStartIndex = lowerCasedString.indexOf(matchedQuery);
+
+  return (
+    <>
+      {string.slice(0, queryStartIndex)}
+      <strong>
+        {string.slice(queryStartIndex, queryStartIndex + matchedQuery.length)}
+      </strong>
+      {string.slice(queryStartIndex + matchedQuery.length)}
+    </>
+  );
+}
+
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: ``,
+      results: [],
+    };
+
+    this.getOrCreateIndex = this.getOrCreateIndex.bind(this);
+    this.search = this.search.bind(this);
+  }
+
+  getOrCreateIndex() {
+    return this.index
+      ? this.index
+      : // Create an elastic lunr index and hydrate with graphql query results
+        Index.load(this.props.searchIndex);
+  }
+
+  search(evt) {
+    const query = evt.target.value;
+    if (query.length > 2) {
```

Comment:
doesn't search until there are 3 or more letters. may want to have a search results page eventually